### PR TITLE
Enabled blobdb on new db only

### PR DESCRIPTION
**Pull request by gfukushima**
## PR Description
This PR changes the behaviour of how we read the blobsdb-enabled flag, it adds a file in the database directory to identify when a database has started from sctrach using blobsdb, after analysis we found that blobdb can save ~5% of disk usage in a full custody node. 
If a database is using SST we will simply ignore the flag and keep using SST files only and no blobdb will be enabled as we found this could cause a bit of backlog to accrue due to SSTs taking a long time to get cleanup and compacted (in some extreme cases it looks like we don't even get to reclaim the disk back).

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about adding a changelog entry, and added one if I deemed necessary.

---

> [!NOTE]
> **Medium Risk**
> Changes on-disk storage layout policy for beacon chain data; mistakes could flip formats or mis-detect fresh vs existing DBs, though detection uses RocksDB CURRENT and tests cover edge cases.
> 
> **Overview**
> **RocksDB BlobDB** for static data columns is no longer driven by the runtime config flag on every open. The factory now resolves an effective mode via `resolveAndPersistBlobDbMode()`, writes it to **`blob-db-mode.txt`**, and applies that value when creating V4/V5/V6 databases.
> 
> **Fresh** databases (no RocksDB `CURRENT` in hot or archive dirs) honor `rocksdbBlobDbEnabled`, including when stale side files remain after wiping `db/`. **Existing** databases keep the recorded mode (or SST if the file is missing for legacy DBs); the CLI flag is ignored so SST stores are not switched to BlobDB mid-life. Invalid mode file contents fail fast instead of defaulting.
> 
> V6 metadata init no longer applies the blob flag there; blob mode is set only at database creation time. Unit tests cover persistence, re-open behavior, legacy DBs, corruption, wiped hot dir, and V5 archive-only survival.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit c5a57c6c0af9bf919379df2bab0cbdced5de01c3. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by rolfyone** [comment]
pointed codex at it. im not really in agreement for the first finding at least, the second and open question are worth having a look at, so sharing below.

Findings
[P2] storage/src/main/java/tech/pegasys/teku/storage/server/VersionedDatabaseFactory.java:463 only checks db/CURRENT, but V4/V5 apply the resulting mode to both db and archive. If the hot DB is wiped while a V4/V5 archive RocksDB remains, a restart with the BlobDB flag can enable BlobDB on an existing SST archive DB, which violates the PR’s “existing DB stays SST” intent. Consider checking archive/CURRENT for V4/V5 or resolving per RocksDB directory.

[P2] storage/src/main/java/tech/pegasys/teku/storage/server/VersionedDatabaseFactory.java:473 uses Boolean.parseBoolean, so any invalid marker content silently becomes false and is then rewritten. Since this marker is authoritative for existing DBs, a truncated or corrupted blob-db-mode.txt could flip an existing BlobDB database to SST instead of failing fast. Parse strictly and throw if the value is not true or false.

Open Question
--Xdata-storage-rocksdb-blob-db-enabled already exists on master from #10470. If that has shipped anywhere, pre-marker BlobDB databases will lack blob-db-mode.txt and this PR will classify them as SST. If it has not shipped, that assumption is fine.

File: storage/src/main/java/tech/pegasys/teku/storage/server/VersionedDatabaseFactory.java [hunk 1]
```diff
@@ -422,6 +423,85 @@ private void saveStorageMode(final StateStorageMode storageMode) {
...
+        blobDbEnabled ? "enabled" : "disabled");
+    return blobDbEnabled;
+  }
+
+  /**
+   * @return whether the RocksDB database directory already contains a database, detected via the
+   *     presence of RocksDB's {@code CURRENT} pointer file
+   */
+  private boolean rocksDbDataExists() {
+    return dbDirectory.toPath().resolve("CURRENT").toFile().exists();
```
[the remark this answers is not in the document]
Comment (gfukushima) [reply storage/src/main/java/tech/pegasys/teku/storage/server/VersionedDatabaseFactory.java]: resolved by the commit below

File: storage/src/main/java/tech/pegasys/teku/storage/server/VersionedDatabaseFactory.java [hunk 2]
```diff
@@ -422,6 +423,92 @@ private void saveStorageMode(final StateStorageMode storageMode) {
...
+
+  /**
+   * @return whether a RocksDB database already exists, detected via the presence of RocksDB's
+   *     {@code CURRENT} pointer file. The archive directory is also checked so that V4/V5 databases
+   *     (which use a separate hot and archive RocksDB) are still treated as existing if only the
+   *     hot directory was wiped. For V6 the archive directory is never created, so that check is a
+   *     no-op.
+   */
+  private boolean rocksDbDataExists() {
+    return rocksDbDataExists(dbDirectory) || rocksDbDataExists(v5ArchiveDirectory);
```
[the remark this answers is not in the document]
Comment (gfukushima) [reply storage/src/main/java/tech/pegasys/teku/storage/server/VersionedDatabaseFactory.java:467]: this scenario doesn't make sense to me.

**Review by rolfyone — APPROVED** [review]
LGTM
